src/utils/EventEmitter.ts

```diff
@@ -8,24 +8,30 @@ function uuidv4(): string {
 
 /** Generic Event Emitter class following a subscribe/publish paradigm. */
 export class EventEmitter<T extends any[]> {
-  subscribers: Record<string, (...args: [...T]) => void | undefined> = {};
+  private subscribers: Map<string, (...args: [...T]) => void | undefined> = new Map();
+
+  constructor() {}
 
   subscribe(s: (...args: [...T]) => void): () => void {
     let uuid = uuidv4();
-    while (this.subscribers[uuid] !== undefined) uuid = uuidv4();
-    this.subscribers[uuid] = s;
+    while (this.subscribers.get(uuid) !== undefined) uuid = uuidv4();
```

Comment:
If you're switching to a Map, you can just switch to a Set and use the fact that all objects have unique identity.